### PR TITLE
Fix non-used device argument in `to_torch` conversion

### LINKED ISSUE #1106: [Bug Report] `NumpyToTorch` wrapper ignores device upon reset
## Bug
[NumpyToTorch](https://github.com/Farama-Foundation/Gymnasium/blob/57136fbaeec340c023ae6939348ac2e7cfe3b3cb/gymnasium/wrappers/numpy_to_torch.py#L102) wrapper ignores `device` when the environment is reset. I created a PR that should fix it.

## Feature request
I would like to further ask if it was possible to change the wrapper's behavior to return tensorized `reward`, `truncated`, `terminated` for non-vectorized environments (or at least add an option to be able to conditionally return their tensor versions)? This is because it is more consistent and more convenient to handle such inputs, for example, when creating tensor buffers:

```python
for i, env_out in enumerate(env_outs[:-1]):
    buffer[i] = torch.cat([buffer[i], env_out.unsqueeze(0)])
```
I tried applying `vectors.numpy_to_torch.NumpyToTorch` wrapper on a non-vector environment which would do the trick, however, there is an assertion that requires the environment to be necessarily a vector env.

### Code example

```shell
import gymnasium as gym
from gymnasium.wrappers.numpy_to_torch import NumpyToTorch

env = NumpyToTorch(gym.make("CartPole-v1"), device="cuda")
print(env.reset()[0].device) # device(type='cpu')
```

### System info

`python=3.11`, `gymnasium=1.0.0a2`

### Additional context

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

### PR DESCRIPTION
**Pull request by mantasu**
# Description

A small fix that propagates the `device` argument when converting from an _Iterable_ of `numpy` or `jax` objects to an _Iterable_ or `torch.Tensor`.

Fixes #1106 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by pseudo-rnd-thoughts — APPROVED** [review]
Thanks for the PR, sadly I don't think we can test this as much as we would wish as the CI server only has a CPU but might be worth adding tests in the future that are ignored by the device unless a gpu is connected
